syntax = "proto3";

package messages.argus.strain.strain_reading;

import "argus/adc.proto";
import "argus/strain/strain_channel.proto";

message StrainReading {
	// Local session from the device that took the reading
	optional int32 local_session = 1;

	// ADC device from which the reading was taken
	adc.AdcDevice adc_device = 2;

	// Identifier for the strain gauge within the ADC device
	strain_channel.StrainChannel strain_channel = 3;

	// Milliseconds since the board's epoch when the reading was recorded
	uint64 recorded_at = 4;

	// Voltage difference measured at the strain wheatstone bridge in millivolts
	float voltage = 5;

	// Strain reading in microstrain
	double strain = 8;
}